api/src/main/java/org/openmrs/validator/DrugOrderValidator.java

```diff
@@ -108,16 +109,33 @@ public void validate(Object obj, Errors errors) {
 			validateFieldsForOutpatientCareSettingType(order, errors);
 			validatePairedFields(order, errors);
 			validateUnitsAreAmongAllowedConcepts(errors, order);
-			validateForRequireDrug(errors);
+			validateForRequireDrug(errors, order);
 		}
 	}
-	
-	private void validateForRequireDrug(Errors errors) {
+
+	private void validateForRequireDrug(Errors errors, DrugOrder order) {
 		//Reject if global property is set to specify a formulation for drug order
 		boolean requireDrug = Context.getAdministrationService().getGlobalPropertyValue(
-		    OpenmrsConstants.GLOBAL_PROPERTY_DRUG_ORDER_REQUIRE_DRUG, false);
-		if (requireDrug) {
-			ValidationUtils.rejectIfEmpty(errors, "drug", "DrugOrder.error.drugIsRequired");
+				OpenmrsConstants.GLOBAL_PROPERTY_DRUG_ORDER_REQUIRE_DRUG, false);
+		OrderService orderService = Context.getOrderService();
+
+
```

Comment:
The reason for not adding that check is because, for a non coded drug the concept need not be DRUG OTHER. This is what is mentioned in the ticket comment right ? 

----> Client is allowed to create a DrugOrder with concept=Ampicillin (e.g. a known concept) and drugNonCoded="Ampicillin 17mg tablet"
the ticket was originally written to say that you can only specify drugNonCoded if concept=DRUG OTHER, but we changed our minds about this